src/StaticResourceHandler/FileLocationRepository.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/**
+ * @see       https://github.com/mezzio/mezzio-swoole for the canonical source repository
+ * @copyright https://github.com/mezzio/mezzio-swoole/blob/master/COPYRIGHT.md
+ * @license   https://github.com/mezzio/mezzio-swoole/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Mezzio\Swoole\StaticResourceHandler;
+
+use InvalidArgumentException;
+
+class FileLocationRepository implements FileLocationRepositoryInterface
+{
+    /**
+     * @var array
+     * Associative array of URI prefixes and directories
+     */
+    private $mappedDocRoots = [];
+
+    /**
+     * Initialize repository with default mapped document roots
+     */
+    public function __construct(array $mappedDocRoots)
+    {
+        // Set up any mapped document roots, validating prefixes and directories
+        foreach ($mappedDocRoots as $prefix => $directory) {
+            if (is_array($directory)) {
+                foreach ($directory as $d) {
+                    $this->addMappedDocumentRoot($prefix, $d);
+                }
+            } else {
+                $this->addMappedDocumentRoot($prefix, $directory);
+            }
```

Comment:
```suggestion
            if (! is_array($directory)) {
                $this->addMappedDocumentRoot($prefix, $directory);
                continue;
            }
            foreach ($directory as $d) {
                $this->addMappedDocumentRoot($prefix, $d);
            }
```

We favor return/continue early, to avoid nesting unnecessarily.